packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+| allocations     | `Allocation[]`                  | Array of funding amounts for participants  |
+| appDefinition   | string                          | Address of contract governing the channel  |
+| appData         | string                          | Encoded initial state of app               |
+| fundingStrategy | `Direct` \|`Ledger` \|`Virtual` | Funding method (type) of channel           |
+
+### Response
+
+// TODO: (HIGH): What about the `SingleChannelResult` and `MultipleChannelResult` documented in the `server-wallet` package?
+
+This method returns a `ChannelResult` as part of the JSON RPC Response, which is an object with the following fields:
+
+| Name                    | Type              | Description                                |
+| ----------------------- | ----------------- | ------------------------------------------ |
+| participants            | `Participant[]`   | Identifying information members of channel |
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+where the channel status can be one of:
+
+TODO: (HIGH) Decided on finalized statuses. What is opening vs. proposed vs. funding vs. running? Is the channel thrown back into the `funding` state when it runs out of money?
+
+- `proposed`: wallet has stored a channel, but no states are signed
+- `opening`: channel has been joined, but is not properly funded
+- `funding`: ???
+- `running`: channel is ready to use
+- `challenging`: channel is in an ongoing dispute
+- `responding`: channel dispute is ongoing, and it is your turn to create a channel update (must send `UpdateChannel` request)
+- `closing`: channel cannot be used, but funds are still locked
+- `closed`: funds have been released from channel
+
+Generally, the `ChannelResult` type is returned when an update to the channel state was made.
+
+### Errors
+
+| Code | Message                | Description                           |
+| ---- | ---------------------- | ------------------------------------- |
+| 1000 | SigningAddressNotFound | Unable to find expected ephemeral key |
+| 1001 | InvalidAppDefinition   | App definition not valid address      |
+| 1002 | UnsupportedToken       | Token in allocations not supported    |
+
+TODO: (HIGH) Verify these are all of the expected errors (what if request times out? what if some participant refuses to join the channel?)
```

Comment:
Note the above list is probably incomplete. More kinds of errors could exist and do not have codes yet.

Reply:
We can try to implement them as completely as possible in the new spec, this would probably be a good place for me to start today